python/ray/tests/test_streaming_generator_4.py

```diff
@@ -191,6 +192,59 @@ async def aio_stream(self):
     ray.get(b.start.remote())
 
 
+@pytest.mark.parametrize("use_asyncio", [False, True])
+def test_cancel(shutdown_only, use_asyncio):
```

Comment:
Can you please elaborate in a comment what we're trying to simulate with this test (got general idea, but want to make sure not missing the details)